test/ci_support/CreateIssueTrackerFromCDashQuery_UnitTests.py

```diff
@@ -0,0 +1,182 @@
+# @HEADER
+# ************************************************************************
+#
+#            TriBTS: Tribal Build, Integrate, and Test System
+#                    Copyright 2013 Sandia Corporation
+#
+# Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+# the U.S. Government retains certain rights in this software.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met:
+#
+# 1. Redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution.
+#
+# 3. Neither the name of the Corporation nor the names of the
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# ************************************************************************
+# @HEADER
+
+import os
+import sys
+import re
+import copy
+import shutil
+import unittest
+import pprint
+
+from FindCISupportDir import *
+import CreateIssueTrackerFromCDashQuery as CIFCDQ
+from GeneralScriptSupport import getCmndOutput, getScriptBaseDir
+from CDashQueryAnalyzeReportUnitTestHelpers import *
+
+g_testBaseDir = getScriptBaseDir()
+
+g_pp = pprint.PrettyPrinter(indent=4)
+
+
+# Base test directory in the build tree
+g_baseTestDir="example_test_failure_github_issue"
+
+
+#############################################################################
+#
+# System-level tests for CreateIssueTrackerFromCDashQuery.py through
+# example_test_failure_github_issue.py
+#
+#############################################################################
+
+
+def example_test_failure_github_issue_test_case(self, testCaseName, cmndLineArgsStr,
+    cdashTestsJsonFile, expectedOutput, expectedIssue,
+    expectedTestsWithIssueTrackersCsv,
+  ):
+  fullTestDir = os.path.join(g_baseTestDir, testCaseName)
```

Comment:
For now, this will support Python 2.7 as long as Python 2.7 is the default Python on SNL systems (i.e. RHEL7).  It is not that hard to support Python 2.7 and 3.x.